pandas/core/dtypes/dtypes.py

```diff
@@ -573,25 +573,112 @@ def _is_boolean(self):
         return is_bool_dtype(self.categories)
 
 
-@register_extension_dtype
-class DatetimeTZDtype(PandasExtensionDtype, ExtensionDtype):
-
+class _DatetimeDtypeBase(PandasExtensionDtype, ExtensionDtype):
     """
     A np.dtype duck-typed class, suitable for holding a custom datetime with tz
     dtype.
 
     THIS IS NOT A REAL NUMPY DTYPE, but essentially a sub-class of
     np.datetime64[ns]
     """
+    _match = None  # type: typing.re.Pattern
     type = Timestamp
     kind = 'M'
     str = '|M8[ns]'
     num = 101
     base = np.dtype('M8[ns]')
     na_value = NaT
+    _cache = {}
+
+    @property
+    def unit(self):
+        """The precision of the datetime data."""
+        return self._unit
+
+    @classmethod
+    def construct_array_type(cls):
+        """
+        Return the array type associated with this dtype
+
+        Returns
+        -------
+        type
+        """
+        from pandas.core.arrays import DatetimeArray
+        return DatetimeArray
+
+    @classmethod
+    def construct_from_string(cls, string):
```

Comment:
I think you need to move this to the sub-class as DatetimeDtype would actually accept this but raise on a construction error (as you can't pass a tz). I would maybe instead just have a unified constructor that accepts ``tz=None`` for naive.